src/windows/disk.rs

```diff
@@ -167,26 +167,32 @@ impl DiskInner {
         self.is_read_only
     }
 
-    pub(crate) fn refresh(&mut self) -> bool {
-        let Some((read_bytes, written_bytes)) = get_disk_io(&self.device_path, None) else {
-            sysinfo_debug!("Failed to update disk i/o stats");
-            return false;
-        };
+    pub(crate) fn refresh_specifics(&mut self, refreshes: DiskRefreshKind) -> bool {
+        if refreshes.kind() && self.type_ == DiskKind::Unknown(-1) {
+            self.type_ = get_disk_kind(&self.device_path, None);
+        }
 
-        self.old_read_bytes = self.read_bytes;
-        self.old_written_bytes = self.written_bytes;
-        self.read_bytes = read_bytes;
-        self.written_bytes = written_bytes;
-
-        if self.total_space != 0 {
-            unsafe {
-                let mut tmp = 0;
-                let lpdirectoryname = PCWSTR::from_raw(self.mount_point.as_ptr());
-                if GetDiskFreeSpaceExW(lpdirectoryname, None, None, Some(&mut tmp)).is_ok() {
-                    self.available_space = tmp;
-                    return true;
-                }
-            }
+        if refreshes.io_usage() {
+            let Some((read_bytes, written_bytes)) = get_disk_io(&self.device_path, None) else {
+                sysinfo_debug!("Failed to update disk i/o stats");
+                return false;
```

Comment:
Considering before it was an "all or nothing", let's not return here (but let keep the error message) and only set the value if the call succeeded.

Reply:
Makes sense; done